R/plumber.R

```diff
@@ -258,52 +264,63 @@ plumber <- R6Class(
     #' be listened on. Note that on most Unix-like systems including Linux and
     #' Mac OS X, port numbers smaller than 1025 require root privileges.
-    #' @param swagger a function that enhances the existing OpenAPI Specification.
-    #' @param debug `TRUE` provides more insight into your API errors.
-    #' @param swaggerCallback a callback function for taking action on the url for swagger page.
+    #' @param debug Deprecated. See `$setDebug()`
+    #' @param swagger Deprecated. See `$setUI(ui)` or `$apiSpecTransform()`
+    #' @param swaggerCallback Deprecated. See `$setUI(callback)`
     #' @details
     #' `port` does not need to be explicitly assigned.
-    #'
-    #' `swagger` should be either a logial or a function . When `TRUE` or a
-    #' function, multiple handles will be added to `plumber` object. OpenAPI json
-    #' file will be served on paths `/openapi.json` and `/swagger.json`. Swagger UI
-    #' will be served on paths `/__swagger__/index.html` and `/__swagger__/`. When
-    #' using a function, it will receive the plumber router as the first parameter
-    #' and current OpenAPI Specification as the second. This function should return a
-    #' list containing OpenAPI Specification.
-    #' See \url{http://spec.openapis.org/oas/v3.0.3}
-    #'
-    #' `swaggerCallback` When set, it will be called with a character string corresponding
-    #' to the swagger UI url. It allows RStudio to open swagger UI when plumber router
-    #' run method is executed using default `plumber.swagger.url` option.
-    #' @examples
-    #' \dontrun{
-    #' pr <- plumber$new
-    #' swagger <- function(pr_, spec) {
-    #'   spec$servers[[1]]$description <- "MyCustomAPISpec"
-    #'   spec
-    #' }
-    #' pr$run(swagger = swagger)
-    #' }
     run = function(
       host = '127.0.0.1',
       port = getOption('plumber.port'),
-      swagger = interactive(),
-      debug = interactive(),
-      swaggerCallback = getOption('plumber.swagger.url', NULL)
+      debug = stop("deprecated"),
+      swagger = stop("deprecated"),
+      swaggerCallback = stop("deprecated")
     ) {
+
       if (isTRUE(private$disable_run)) {
         stop("Plumber router `$run()` method should not be called while `plumb()`ing a file")
       }
 
-      port <- findPort(port)
+      # Legacy support for RStudio pro products.
+      # Checks must be kept for >= 2 yrs after plumber v1.0.0 release date
+      if (!missing(debug)) {
+        message("`$run(debug)` has been deprecated in v1.0.0 and will be removed in a coming release. Please use `$setDebug(debug)`")
+        self$setDebug(debug)
+      }
+      if (!missing(swagger)) {
+        if (is.function(swagger)) {
+          # between v1.0.0 and v0.4.6
+          message("`$run(swagger)` has been deprecated in v1.0.0 and will be removed in a coming release. To alter the swagger spec, please use `$apiSpecTransform(api_fun)`")
+          self$apiSpecTransform(swagger)
```

Comment:
```suggestion
          self$apiSpecTransform(swagger)
          self$setUI(TRUE)
```

Reply:
Opting against this as it is enabled by default and they may have set a UI sometime earlier.  Keep the message about `swagger` can not be a function.